src/Plugin/formatter/FormatterHalJson.php

```diff
@@ -186,6 +190,21 @@ protected function extractFieldValues($rows) {
       if (!$rows instanceof ResourceFieldCollectionInterface) {
         throw new InternalServerErrorException('Inconsistent output.');
       }
+
+      // This feels a bit awkward, but if the result is going to be cached, it
+      // pays off the extra effort of generating the whole resource entity. That
+      // way we can get a different field set with the previously cached entity.
+      // If the entity is not going to be cached, then avoid generating the
+      // field data altogether.
+      $limit_fields = $rows->getLimitFields();
```

Comment:
why do even need this check - can't we remove this whole block - and simply wait for the caching [few](https://github.com/RESTful-Drupal/restful/pull/660/files#diff-bf10e1e10ec909f1a0ad72002bb6dd55R221) lines below?